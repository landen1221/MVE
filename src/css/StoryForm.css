.StoryForm {
  margin-top: 6em;
}

.StoryForm .MuiButton-containedPrimary {
  background-color: rgb(0, 99, 0);
  margin-left: 15px;
}

.StoryForm .MuiButton-containedSecondary {
  background-color: rgb(202, 0, 0);
}

.StoryForm .error {
  margin: 0;
  color: rgb(191, 0, 0);
  font-style: italic;
  font-size: 0.9em;
}

.MuiInputBase-root {
  width: 35em;
}

#age-field {
  text-align: center;
}

.MuiFormControl-root {
  margin-left: 0;
}

.MuiFormGroup-root {
  align-content: center;
}

.StoryForm .MuiFormControlLabel-root {
  margin: 0.3em;
}

.StoryForm .MuiIconButton-label {
  margin-right: 0.5em;
}

.StoryForm .MuiFormLabel-root {
  margin-bottom: 0.5em;
}

.StoryForm .MuiRadio-root {
  margin: 0;
  padding: 0;
}

.StoryForm h3 {
  margin-bottom: 8px;
}

#story-label {
  text-align: center;
  color: rgba(0, 0, 0, 0.54);
  padding: 0;
  font-size: 1rem;
  font-weight: 400;
  line-height: 0.7;
  letter-spacing: 0.00938em;
  margin-bottom: 0.5rem;
}

#story-helper {
  margin-top: 3px;
  margin-bottom: 2em;
  font-family: Roboto, Helvetica, Arial, sans-serif;
  font-size: 14px;
  color: rgba(0, 0, 0, 0.755);
}

#username-helper {
  margin-top: 3px;
  margin-bottom: 15px;
  font-family: Roboto, Helvetica, Arial, sans-serif;
  font-size: 12px;
}

#center-recaptcha {
  width: 19em;
  margin: auto;
  margin-top: 15px;
  text-align: center;
}
