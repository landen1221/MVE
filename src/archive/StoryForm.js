import { useState } from "react";
import { Link, useHistory } from "react-router-dom";
import TextField from "@material-ui/core/TextField";
import Radio from "@material-ui/core/Radio";
import RadioGroup from "@material-ui/core/RadioGroup";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import FormControl from "@material-ui/core/FormControl";
import FormLabel from "@material-ui/core/FormLabel";
import Button from "@material-ui/core/Button";
import MenuItem from "@material-ui/core/MenuItem";
import "../css/StoryForm.css";
import UsernameGenerator from "username-generator";
import MVEAPI from "../api";

const StoryForm = ({ vaccines }) => {
  let history = useHistory();
  const initialData = {
    username: UsernameGenerator.generateUsername(),
    vaccine: "",
    satisfied: "",
    age: 0,
    gender: "",
    story: "",
  };

  const [formData, setFormData] = useState(initialData);
  // const [errors, setErrors] = useState({});

  const handleSubmit = (evt) => {
    evt.preventDefault();

    async function addStory() {
      let added = await MVEAPI.postStory(formData);
      console.log(added);
    }

    addStory();

    console.log("Success. User data:");
    // console.log(storyData);
    history.push(
      formData.vaccine === "COVID" ? `/covid` : `/vaccine/${formData.vaccine}`
    );
  };

  const handleChange = (evt) => {
    const { name, value } = evt.target;

    setFormData((f) => ({
      ...f,
      [name]: value,
    }));
  };

  const extraVaccineRow = (
    <FormControl component="fieldset">
      <FormLabel component="legend">Glad I got the Vaccine:*</FormLabel>
      <RadioGroup
        aria-label="satisfied"
        name="satisfied"
        value={formData.satisfied}
        onChange={handleChange}
        required
      >
        <FormControlLabel value="true" control={<Radio />} label="Yes" />
        <FormControlLabel value="false" control={<Radio />} label="No" />
      </RadioGroup>
    </FormControl>
  );

  const extraCovidRow = (
    <FormControl component="fieldset">
      <FormLabel component="legend">Intensity of Illness</FormLabel>
      <RadioGroup
        aria-label="satisfied"
        name="satisfied"
        value={formData.satisfied}
        onChange={handleChange}
      >
        <FormControlLabel
          value="No Big Deal"
          control={<Radio />}
          label="No Big Deal"
        />
        <FormControlLabel value="Mild" control={<Radio />} label="Mild" />
        <FormControlLabel
          value="Moderate"
          control={<Radio />}
          label="Moderate"
        />
        <FormControlLabel value="Severe" control={<Radio />} label="Severe" />
      </RadioGroup>
    </FormControl>
  );

  return (
    <div className="StoryForm">
      <h3>Submit Story about your COVID or Vaccine experience:</h3>
      <p id="story-helper">
        *Story should be 1st hand experience (not something you heard/read)
        <br />
        *This is a public forum. Please don't include personal/private
        information
      </p>
      <form>
        <TextField
          label="Username"
          variant="outlined"
          name="username"
          value={formData.username}
          helperText="*You may keep auto-generated username (Save username to find your post later)"
          onChange={handleChange}
          required
        />
        <br />
        <br />
        <TextField
          label="Age (Optional)"
          placeholder="Optional"
          variant="outlined"
          name="age"
          onChange={handleChange}
        />
        <br /> <br />
        <TextField
          id="select-gender"
          select
          label="Gender (optional)"
          value={formData.gender}
          onChange={handleChange}
          name="gender"
          variant="outlined"
        >
          <MenuItem key="male" value="male">
            Male
          </MenuItem>
          <MenuItem key="female" value="female">
            Female
          </MenuItem>
          <MenuItem key="other" value="other">
            Other
          </MenuItem>
        </TextField>
        <br />
        <br />
        <TextField
          id="select-vaccine"
          select
          label="Select COVID or Vaccine:"
          value={formData.vaccine}
          name="vaccine"
          onChange={handleChange}
          variant="outlined"
          required
        >
          <MenuItem key={"covid"} value={"covid"} name={"covid"}>
            COVID
          </MenuItem>
          {Object.entries(vaccines).map(([dbName, siteName]) => (
            <MenuItem key={dbName} value={dbName} name={dbName}>
              {siteName}
            </MenuItem>
          ))}
        </TextField>
        <br />
        <br />
        {formData.vaccine === "covid" ? extraCovidRow : extraVaccineRow}
        <br />
        <p id="story-label">My Story:*</p>
        <TextField
          placeholder="After my 2nd vaccine I didn't feel great for a few days, but I'm glad I got it because..."
          id="story-textarea"
          variant="outlined"
          multiline
          rows={5}
          rowsMax={Infinity}
          name="story"
          value={formData.story}
          onChange={handleChange}
          required
        />
        <br />
        <br />
        <Link to="/">
          <Button
            variant="contained"
            color="secondary"
            className="VaccineButton"
          >
            Cancel
          </Button>
        </Link>
        <Button
          variant="contained"
          color="primary"
          className="VaccineButton"
          onClick={handleSubmit}
        >
          Submit
        </Button>
      </form>
    </div>
  );
};

export default StoryForm;
