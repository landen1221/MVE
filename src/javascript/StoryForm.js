import { Link, useHistory } from "react-router-dom";
import TextField from "@material-ui/core/TextField";
import Radio from "@material-ui/core/Radio";
import RadioGroup from "@material-ui/core/RadioGroup";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import FormControl from "@material-ui/core/FormControl";
import FormLabel from "@material-ui/core/FormLabel";
import Button from "@material-ui/core/Button";
import MenuItem from "@material-ui/core/MenuItem";
import "../css/StoryForm.css";
import UsernameGenerator from "username-generator";
import MVEAPI from "../api";

import { useFormik } from "formik";
import * as Yup from "yup";

const StoryForm = ({ vaccines, setCurrStory }) => {
  let history = useHistory();

  const formik = useFormik({
    initialValues: {
      username: `${UsernameGenerator.generateUsername()}${Math.floor(
        Math.random() * 100
      )}`,
      vaccine: "",
      satisfied: "",
      age: undefined,
      gender: "",
      story: "",
      // fingerprint: "",
    },

    validationSchema: Yup.object({
      username: Yup.string()
        .max(35, <p className="error">Must be 35 characters or less</p>)
        .required(<p className="error">Username Required</p>),
      vaccine: Yup.string().required(
        <p className="error">Must select COVID or Vaccine</p>
      ),
      satisfied: Yup.string().required(<p className="error">Required</p>),
      age: Yup.number()
        .min(
          14,
          <p className="error">Unfortunately you must be at least 14 to post</p>
        )
        .max(
          120,
          <p className="error">Invalid age. Enter age or leave blank</p>
        )
        .nullable(),
      gender: Yup.string(),
      story: Yup.string()
        .min(10, <p className="error">Story must be at least 10 characters.</p>)
        .required(
          <p className="error">
            This site is built by your stories. Please share your experience.
          </p>
        ),
    }),
    onSubmit: (values) => {
      console.log(values);
      async function addStory() {
        await MVEAPI.postStory(values);
      }
      addStory();
      setCurrStory(values);

      history.push(
        values.vaccine === "COVID" ? `/covid` : `/vaccine/${values.vaccine}`
      );
    },
  });

  const extraVaccineRow = (
    <FormControl component="fieldset">
      <FormLabel component="legend">
        <u>Glad I got the Vaccine:</u>
      </FormLabel>
      <RadioGroup
        aria-label="satisfied"
        name="satisfied"
        value={formik.values.satisfied}
        onBlur={formik.handleBlur}
        onChange={formik.handleChange}
      >
        <FormControlLabel value="Yes" control={<Radio />} label="Yes" />
        <FormControlLabel value="No" control={<Radio />} label="No" />
      </RadioGroup>
    </FormControl>
  );

  const extraCovidRow = (
    <FormControl component="fieldset">
      <FormLabel component="legend">
        <u>Intensity of Illness</u>
      </FormLabel>
      <RadioGroup
        aria-label="satisfied"
        name="satisfied"
        value={formik.values.satisfied}
        onBlur={formik.handleBlur}
        onChange={formik.handleChange}
      >
        <FormControlLabel
          value="No Big Deal"
          control={<Radio />}
          label="No Big Deal"
        />
        <FormControlLabel value="Mild" control={<Radio />} label="Mild" />
        <FormControlLabel
          value="Moderate"
          control={<Radio />}
          label="Moderate"
        />
        <FormControlLabel value="Severe" control={<Radio />} label="Severe" />
      </RadioGroup>
    </FormControl>
  );

  return (
    <div className="StoryForm">
      <h3>Submit Story about your COVID or Vaccine experience:</h3>
      <p id="story-helper">
        Story should be 1st hand experience (not something you heard/read){" "}
        <br />
        This is a public forum. Don't include personal/private information
      </p>

      <form onSubmit={formik.handleSubmit}>
        <TextField
          label="Username"
          variant="outlined"
          name="username"
          helperText="*You may keep auto-generated username (Save username to find your post later)"
          value={formik.values.username}
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
        />
        {formik.touched.username && formik.errors.username ? (
          <div>{formik.errors.username}</div>
        ) : null}
        <br />
        <br />

        <TextField
          id="select-vaccine"
          select
          label="Select COVID or Vaccine:"
          name="vaccine"
          variant="outlined"
          value={formik.values.vaccine}
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
        >
          <MenuItem key={"covid"} value={"covid"} name={"covid"}>
            COVID
          </MenuItem>
          {Object.entries(vaccines).map(([dbName, siteName]) => (
            <MenuItem key={dbName} value={dbName} name={dbName}>
              {siteName}
            </MenuItem>
          ))}
        </TextField>
        {formik.touched.vaccine && formik.errors.vaccine ? (
          <div>{formik.errors.vaccine}</div>
        ) : null}
        <br />
        <br />

        {formik.values.vaccine === "covid" ? extraCovidRow : extraVaccineRow}
        {formik.touched.satisfied && formik.errors.satisfied ? (
          <div>{formik.errors.satisfied}</div>
        ) : null}
        <br />
        <br />
        <TextField
          id="age-field"
          label="Age (Optional)"
          placeholder="Optional"
          variant="outlined"
          name="age"
          value={formik.values.age}
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
        />
        {formik.touched.age && formik.errors.age ? (
          <div>{formik.errors.age}</div>
        ) : null}
        <br />
        <br />
        <TextField
          id="select-gender"
          select
          label="Gender (optional)"
          name="gender"
          variant="outlined"
          value={formik.values.gender}
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
        >
          <MenuItem key="Male" value="Male">
            Male
          </MenuItem>
          <MenuItem key="Female" value="Female">
            Female
          </MenuItem>
          <MenuItem key="Other" value="Other">
            Other
          </MenuItem>
        </TextField>
        {formik.touched.gender && formik.errors.gender ? (
          <div>{formik.errors.gender}</div>
        ) : null}
        <br />
        <br />

        <p id="story-label">My Story:*</p>
        <TextField
          placeholder="After my 2nd vaccine I didn't feel great for a few days, but I'm glad I got it because..."
          id="story-textarea"
          variant="outlined"
          multiline
          rows={5}
          rowsMax={Infinity}
          name="story"
          value={formik.values.story}
          onChange={formik.handleChange}
        />
        {formik.touched.story && formik.errors.story ? (
          <div>{formik.errors.story}</div>
        ) : null}

        <br />
        <br />

        <Link to="/">
          <Button
            variant="contained"
            color="secondary"
            className="VaccineButton"
          >
            Cancel
          </Button>
        </Link>
        <Button
          variant="contained"
          color="primary"
          className="VaccineButton"
          type="submit"
        >
          Submit
        </Button>
      </form>
    </div>
  );
};
export default StoryForm;
