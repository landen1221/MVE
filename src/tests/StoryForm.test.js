import React from "react";
import { fireEvent, render } from "@testing-library/react";
import StoryForm from "../javascript/StoryForm";
import { vaccines } from "./globalTestVariables";
import { MemoryRouter } from "react-router";
import { act } from "react-dom/test-utils";
import { screen } from "@testing-library/react";

// Props: vaccines, setCurrStory

// FIXME: How to add setCurrStory

it("renders without crashing", () => {
  render(
    <MemoryRouter>
      <StoryForm vaccines={vaccines} />
    </MemoryRouter>
  );
});

// it("matches snapshot", function () {
//   const { asFragment } = render(
//     <MemoryRouter>
//       <StoryForm vaccines={vaccines} />
//     </MemoryRouter>
//   );
//   expect(asFragment()).toMatchSnapshot();
// });

// describe("complete form", () => {
//   describe("with valid inputs", () => {
//     it("calls the onSubmit function", async () => {
//       const mockOnSubmit = jest.fn();
//       const { getByLabelText, getByRole } = render(
//         <MemoryRouter>
//           <StoryForm vaccines={vaccines} onSubmit={mockOnSubmit} />
//         </MemoryRouter>
//       );
//       await act(async () => {
//         // for covid replace 'Glad I got the Vaccine:' with  'Intensity of Illness:'
//         fireEvent.change(getByLabelText("Username:"), {
//           target: { value: "testUser100" },
//         });
//         fireEvent.change(getByLabelText("Select COVID or Vaccine:"), {
//           target: { value: "pfizer" },
//         });
//         fireEvent.change(getByLabelText("Glad I got the Vaccine:"), {
//           target: { value: "Yes" },
//         });
//         fireEvent.change(getByLabelText("Age (Optional):"), {
//           target: { value: 32 },
//         });
//         fireEvent.change(getByLabelText("Gender (optional):"), {
//           target: { value: "Male" },
//         });
//         fireEvent.change(getByLabelText("My Story:"), {
//           target: {
//             value:
//               "Didn't feel well for a couple of days but now I feel great!",
//           },
//         });
//       });
//       await act(async () => {
//         fireEvent.click(getByRole("submit"));
//       });

//       expect(mockOnSubmit).toHaveBeenCalled();
//     });
//   });
// });
