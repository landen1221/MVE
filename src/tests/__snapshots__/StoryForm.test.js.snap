// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="StoryForm"
  >
    <h3>
      Submit Story about your COVID or Vaccine experience:
    </h3>
    <p
      id="story-helper"
    >
      Story should be 1st hand experience (not something you heard/read) 
      <br />
      This is a public forum. Don't include personal/private information
    </p>
    <form>
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
          data-shrink="true"
        >
          Username:
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            name="username"
            type="text"
            value="scripturesweaty77"
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-12 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-14 PrivateNotchedOutline-legendNotched-15"
            >
              <span>
                Username:
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-filled"
        >
          *You may keep auto-generated username (Save username to find your post later)
        </p>
      </div>
      <br />
      <br />
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
          for="select-vaccine"
          id="select-vaccine-label"
        >
          Select COVID or Vaccine:
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby="select-vaccine-label select-vaccine"
            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
            id="select-vaccine"
            role="button"
            tabindex="0"
          >
            <span>
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput"
            name="vaccine"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-12 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-14"
            >
              <span>
                Select COVID or Vaccine:
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <br />
      <br />
      <fieldset
        class="MuiFormControl-root"
      >
        <legend
          class="MuiFormLabel-root"
        >
          <u>
            Glad I got the Vaccine:
          </u>
        </legend>
        <div
          aria-label="satisfied"
          class="MuiFormGroup-root"
          role="radiogroup"
        >
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-16 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-19"
                  name="satisfied"
                  type="radio"
                  value="Yes"
                />
                <div
                  class="PrivateRadioButtonIcon-root-20"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-21"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              Yes
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-16 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-19"
                  name="satisfied"
                  type="radio"
                  value="No"
                />
                <div
                  class="PrivateRadioButtonIcon-root-20"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-21"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              No
            </span>
          </label>
        </div>
      </fieldset>
      <br />
      <br />
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
          for="age-field"
          id="age-field-label"
        >
          Age (Optional):
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            id="age-field"
            name="age"
            placeholder="Optional"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-12 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-14"
            >
              <span>
                Age (Optional):
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <br />
      <br />
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
          for="select-gender"
          id="select-gender-label"
        >
          Gender (optional):
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby="select-gender-label select-gender"
            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
            id="select-gender"
            role="button"
            tabindex="0"
          >
            <span>
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput"
            name="gender"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-12 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-14"
            >
              <span>
                Gender (optional):
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <br />
      <br />
      <p
        id="story-label"
      >
        My Story:
      </p>
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
        >
          <textarea
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
            id="story-textarea"
            name="story"
            placeholder="After my 2nd vaccine I didn't feel great for a few days, but I'm glad I got it because..."
            rows="5"
          />
          <textarea
            aria-hidden="true"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
            readonly=""
            style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
            tabindex="-1"
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-12 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-13"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <br />
      <br />
      <a
        href="/"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained VaccineButton MuiButton-containedSecondary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Cancel
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </a>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained VaccineButton MuiButton-containedPrimary"
        role="submit"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Submit
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>
</DocumentFragment>
`;
